app/(public)/getProducts/page.tsx

```diff
@@ -0,0 +1,74 @@
+'use client'
+
+import CustomInput from "@/app/components/GetProducts/CustomInput";
+import ProductProperty from "@/app/components/GetProducts/ProductProperty";
+import { Product } from "@/app/components/GetProducts/definition";
+import { apiRequest } from "@/app/utils/apiRequest";
+import { useState } from "react";
+
+export default function getProducts() {
+    const [productName, setProductName] = useState('');
+    const [itemCode, setItemCode] = useState('');
+    const [realItemCode, setRealItemCode] = useState('');
+    const [quantity, setQuantity] = useState(0);
+    const [products, setProducts] = useState([]);
+
+
+    async function handlePost(e: React.FormEvent<HTMLFormElement>) {
+        e.preventDefault();
+        const data = await apiRequest('/api/getProducts', 'POST', { productName, itemCode, quantity })
+        setProducts(data.rows);
+    }
+
+    async function handleGet(e: React.MouseEvent<HTMLButtonElement>) {
+        e.preventDefault();
+        const data = await apiRequest(`/api/getProducts?itemCode=${realItemCode}`, 'GET')
+        setProducts(data);
+    }
+
+    async function handleGetAll(e: React.MouseEvent<HTMLButtonElement>) {
+        e.preventDefault();
+        const data = await apiRequest('/api/getProducts', 'GET')
+        setProducts(data);
+    }
+
+    const isPostValid = itemCode !== '' && productName !== '' && quantity > 0;
+    const isGetValid = realItemCode !== '';
+
+
+    return (
+        <div className="flex flex-wrap justify-center place-items-center text-white">
+            <div className="w-full flex flex-col items-center justify-center mt-20">
+                <form className="h-40 w-[700px] border border-black flex justify-center items-center text-black my-2" onSubmit={handlePost}>
+                    <CustomInput onChange={(e) => setProductName(e.target.value)} placeholder="Product Name" />
+                    <CustomInput onChange={(e) => setItemCode(e.target.value)} placeholder="Item Code" />
+                    <CustomInput onChange={(e) => setQuantity(parseInt(e.target.value))} placeholder="Quantity" />
+                    <button className={`w-20 h-10 bg-black text-white rounded-sm ${isPostValid ? 'cursor-pointer' : 'cursor-not-allowed'}`} disabled={!isPostValid} type="submit">POST</button>
+                </form>
+                <form className="h-40 w-[700px] border border-black flex justify-center items-center text-black my-2">
+                    <CustomInput onChange={(e) => setRealItemCode(e.target.value)} placeholder="Real Item Code" />
+                    <button className={`w-20 h-10 bg-black text-white rounded-sm ${isGetValid ? 'cursor-pointer' : 'cursor-not-allowed'} mr-2`} onClick={handleGet} disabled={!isGetValid}>GET</button>
+                    <button className="w-20 h-10 bg-black text-white rounded-sm cursor-pointer" onClick={handleGetAll}>GET ALL</button>
+                </form>
```

Comment:
Two forms...hmmm. Possibly another component. Separating it will make the className easier to manage.